src/main/java/de/bonndan/nivio/ProcessingFinishedEvent.java

```diff
@@ -12,7 +13,11 @@
 
     private final Landscape landscape;
 
-    public ProcessingFinishedEvent(Object source, Landscape landscape) {
+    /**
+     * @param source TODO is this a misuse of event source? We store the LandscapeDescription instead of the triggerer
```

Comment:
Don't let @Matthimatiker hear this :smile:

What bothers me is that we carry the final graph "Landscape" with the processing events and now suddenly introduce the source ("LandscapeDescription") by using an existing field that has always been referencing the triggering instance. However, perhaps your view is right and the triggering instance is of no concern in our context anyways.

Reply:
Don't worry, I won't tell him!

Well, I just don't have a complete overview on our application's architecture yet. Of course, if we used to do it differently and now change only one pattern, this might not be consistent anymore - if that is the concern?

So, do I understand correctly: we carry the final `Landscape` around while processing. Meaning that we have an object which we construct in the process? And now we have the source, which we also can pass around. Does this mean that we could get rid of carying the `Landscape`around?

Reply:
No the lanscape is the final graph - the only thing which is of interest to us usually. In the case of an event which triggers indexing (re-reading the whole sources), we also need the configuration that is the input for indexing, since we want to observe it for future changes.